### DISCUSSION TITLE
Hard maximum | Time

### DISCUSSION THREAD

**Discussion by vlaksi** [libuv/libuv#3910] [Q&A]
Hey folks,

I saw this on Node doc: 

"To prevent the poll phase from starving the event loop, [libuv](https://libuv.org/) (the C library that implements the Node.js event loop and all of the asynchronous behaviors of the platform) also has a hard maximum (system dependent) before it stops polling for more events." and I tried to check how big is **hard maximum** time, but I can not find a normal answer or I can not belive what I found or I do not understand smth.

Ie. I found out hard maximum values for different operating systems:

- Linux: typically around 2^31-1 milliseconds (about 24.8 days)
- macOS: typically around 2^31-2 milliseconds (about 49.7 days)
- Windows: typically around 2^31-1 milliseconds (about 24.8 days)

I do not understand how it is needed to wait so much before the event loop moves to the other phases. I thought it will be ~0.02 milliseconds and then move to another iteration of the event loop...

**Comment by vtjnash** [comment]
It is not a time based factor, because that varies by hardware, but a simple count of read or write operations performed. Also, it definitely should be assumed to be longer than 0.02 ms. The tick rate on most kernels is around 100Hz (or usually even slower for special high performance machine). The maximum is intended to reduce performance to a given consumer, to the benefit of fairness to all consumers.
